### PR TITLE
Validation model

### PR DESCRIPTION
**Pull request by Ngufor-emmanuella**
###  in thismilestone of the project I Implementations
Add the following validations:

For the User model:
Name mu[s](https://github.com/soulemanou-software)t not be blank.
PostsCounter must be an integer greater than or equal to zero.
For the Post model:
Title must not be blank.
Title must not exceed 250 characters.
CommentsCounter must be an integer greater than or equal to zero.
LikesCounter must be an integer greater than or equal to zero.
Add unit specs for all of your models' methods and validations.

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+class Post < ApplicationRecord
```
Comment (Alfredbis29) [inline app/models/post.rb:1]: Good job on making all validation required to this project :wink:  :clap:

File: test/channels/application_cable/connection_test.rb [hunk 2]
```diff
@@ -1,4 +1,4 @@
-require "test_helper"
+require 'test_helper'
```
Comment (Alfredbis29) [inline test/channels/application_cable/connection_test.rb]: - [ ] As required for this milestone of the project, it is crucial for the project to include unit tests for the User, Post, Comment, and Like models. Unfortunately, it appears that these tests have not yet been added to the project. :smile:  

I understand that writing tests can be challenging, but they are a critical component of ensuring the quality and reliability of the code. I would like to request that you make the necessary changes to include the unit tests as soon as possible. for your project to match all requirements :wink:  kindly add thoses tests to your project then ask for another review , I'm sure you are going to make it :muscle:  :wink:

**Review by Alfredbis29 — CHANGES_REQUESTED** [review]
### Hi @Ngufor-emmanuella ,
 You did a great job so far 👏🏼 👏🏼 There are some issues that you still need to work on to go to the next project but you are almost there!
 
 
 ### To highlight 💪🏻
 * All test files is placed in a different directory. :heavy_check_mark:
* Gitflow was correctly used. :heavy_check_mark:
:+1:  Good job!

## STATUS: CHANGES REQUESTED 
_Please Check the comments under the review and make all required changes_
 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

 Cheers and Happy coding! :clap: :clap: :clap: :computer: Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please ping me @alfredbis29 when you comment so I can receive the notification.
 
 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by KDlamini** [comment]
## Status: Invalid 🔴 

### Hi @Ngufor-emmanuella 🙋‍♀️,  

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request
 
You have submitted a project, that is failing to run tests because you are missing important gems in your gemfile.

- Please include the `rspec-core` and `rspec-rails`  gems in gemfile so that users can be able to run tests.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏
 
Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me @KDlamini  in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by Ngufor-emmanuella** [comment]
thank you @KDlamini for the review... please a quick question for clarification ...am i adding them manually or am I running the code gem install rspec please

**Comment by KDlamini** [comment]
> thank you @KDlamini for the review... please a quick question for clarification ...am i adding them manually or am I running the code gem install rspec please

@Ngufor-emmanuella Please add them to your Gemfile.

**Comment by Ngufor-emmanuella** [comment]
Hello You requested me to install the rspec and rspec rails gems, however, these gems are not part of the project requirements for this milestone. The gems are used to do the unit testing of the models, the project requirement doesn't require testing yet. Please look into this.thanks

File: Gemfile [hunk 3]
```diff
@@ -50,12 +50,12 @@ gem "bootsnap", require: false
 
 group :development, :test do
   # See https://guides.rubyonrails.org/debugging_rails_applications.html#debugging-with-the-debug-gem
-  gem "debug", platforms: %i[ mri mingw x64_mingw ]
+  gem 'debug', platforms: %i[mri mingw x64_mingw]
 end
```
Comment (OLIPLICHE) [inline Gemfile]: - [ ]  **Please follow this suggestion and your test will work:**
- Add **gem 'rspec-rails'** in `group :development, :test do`
- run **bundle install** in your terminal
- Then run  **rspec spec/models** to test if the codes has passed
- Then submit for re-review.
- All the best.
-  Refer to this 👉[  Article](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/Model_specs_lesson_v1_1.md) to learn more

**Review by OLIPLICHE — CHANGES_REQUESTED** [review]
Hi @Ngufor-emmanuella , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
- Linter is green (the last commit has a green check mark (✅)) 👏
- README file is descriptive 👏
- Commit messages are meaningful 👏 👏

 ## Required Changes ♻️
- Hi, I would advise you to provide unit tests as was mentioned by the previous review. It is a part of the project requirements.
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: Gemfile (hunk 3 above)
Comment (Ngufor-emmanuella) [reply Gemfile]: thank you

**Comment by OLIPLICHE** [comment]
- Hi, I will suggest you not to push code when your PR is under review. it is a bad practice and can create confusion. hence it is going to delay your progress. 👍 Make sure to Clean the mistake in Linter

**Review by danielufeli — APPROVED** [review]
Hi @Ngufor-emmanuella ,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

ChopApproveCopApproveGIF

## Highlights
- Nice commits messages✔️
- Clean Code ✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @danielufeli in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._ate_comments_counter`

**Comment by Ngufor-emmanuella** [comment]
thank you so much

**Pull request merged**